src/reader.rs

```diff
@@ -38,10 +34,20 @@ struct NamespaceData {
     delegates: std::vec::Vec<(u32, u32)>,
 }
 
-pub struct Namespace<'a> {
-    reader: &'a Reader,
-    name: &'a str,
-    types: &'a NamespaceData,
+impl<'a> Iterator for NamespaceIterator<'a> {
+    type Item = Namespace<'a>;
+    fn next(&mut self) -> Option<Self::Item> {
```

Comment:
This iterator seems like it can be accomplished using `map` on iterator. Is there a need for this custom iterator?

Reply:
I'll check.